src/reducers/challenge-reducer.ts

```diff
@@ -39,16 +41,26 @@ type InitializeAction = {
   }
 };
 
-type ChallengeAction = InitializeAction | ToggleAction | CheckAction;
+type AnswerAction = {
+  type: 'SET_ANSWER';
+  payload: {
+    challengeId: string;
+    answer: string;
+  }
+};
```

Comment:
It's feeling very repetitive to create an action type, an action creator, and a function in the Mutators that dispatches the action type. It'd be nice if I could define the actions as classes and get ride of at least the action type/action creator function pair. It feels like it ought to be possible to make TS figure out the correct shape of the payload in the reducers. I'll have to try it later.